inst/tests/tests.Rraw

```diff
@@ -6031,6 +6031,31 @@ thisDT <- copy(DT)[, c("aaa", "b") := 2]
 test(1419.58, indices(thisDT), c("a", "ab"))
 test(1419.59, allIndicesValid(thisDT), TRUE)
 
+## setkey on same col as index before
+DT <- data.table(a =c(4,1,3,9,2,1,8,7,6,5),
+                 aaa = c(1,1,2,2,2,1,1,2,2,2))
+setindex(DT, a)
+test(1419.60, allIndicesValid(DT), TRUE)
+setindex(DT, NULL)
+setkey(DT, a)
+test(1419.61, DT$a, c(1,1,2,3,4,5,6,7,8,9))
+setkey(DT, NULL)
+setindex(DT, a)
+test(1419.62, setkey(DT, a, verbose=TRUE), output="using existing index for a")
+
+# check setkey incase of existing multiple indexes
+DT <- data.table(a = c(3,3,4,4,5,6,1,1,7,2,2),
+                 aaa = c(1,1,2,2,2,1,1,2,2,2),
+                 bbb = c(1,1,2,0,1,1,1,0,1,1))
+setkey(DT, a)
+test(1419.63, DT$a, c(1,1,2,2,3,3,4,4,5,6,7))
+setkey(DT, NULL)
+test(1419.64, setkey(DT, a, verbose=TRUE), output="forder took")
+setkey(DT, NULL)
+setindex(DT, a)
+setindex(DT, aaa)
+test(1419.65, allIndicesValid(DT), TRUE)
+test(1419.66, setkey(DT, aaa, verbose=TRUE), output="using existing index for aaa")
```

Comment:
This doesn't check the result of setkey is correct.  Please see previous comments where that was requested :  https://github.com/Rdatatable/data.table/pull/3582#discussion_r287548138 and https://github.com/Rdatatable/data.table/pull/3582#discussion_r287548675

Reply:
thanks for chipping in. I wasn't fully aware of `test` function parameters. Now I understand it better. Also, I will try to fix my local environment (I still don't know how am I gonna do that) to avoid stupid errors in the PR.